bench: replace stack allocation with dynamic allocation

navyansh007 opened the pull request:
fix(math/strided/special/dfloor): replace stack allocation with dynamic allocation in benchmark

Replace fixed-size array declarations with dynamic memory allocation in the dfloor benchmark to prevent segmentation faults when processing large input sizes. The benchmark previously failed with segmentation faults for arrays with 1,000,000 elements due to stack memory limitations.

  Changes:
  - Replace stack-allocated arrays with malloc/free for x and y arrays
  - Add proper error handling for memory allocation failures
  - Add memory cleanup to prevent memory leaks
  - Add error checking in main benchmark loop

  Fixes: #7224

-   [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

navyansh007 committed:
fix(math/strided/special/dfloor): replace stack allocation with dynamic allocation in benchmark

  Replace fixed-size array declarations with dynamic memory allocation in
  the dfloor benchmark to prevent segmentation faults when processing
  large input sizes. The benchmark previously failed with segmentation
  faults for arrays with 1,000,000 elements due to stack memory limitations.

  Changes:
  - Replace stack-allocated arrays with malloc/free for x and y arrays
  - Add proper error handling for memory allocation failures
  - Add memory cleanup to prevent memory leaks
  - Add error checking in main benchmark loop

  Fixes: #7224

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented:
@navyansh007 Would you mind updating this PR to match the dynamic memory allocation changes in this commit: https://github.com/stdlib-js/stdlib/commit/2c9f492d8e15e860aa56f9d3b5b3845191480700? Cheers!

navyansh007 committed:
Merge branch 'stdlib-js:develop' into dfloor-benchmark-segmentation-fix

navyansh007 committed:
refactor: replaced previous malloc statements to match new code style

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

navyansh007 commented:
Hey @kgryte,
I have updated the `malloc` statements in the code and changed them to match the way they are written in the commit you mentioned. I request you to have a look at that. Feel free to let me know if any further changes are required.

kgryte committed:
bench: make consistent with elsewhere

Signed-off-by: Athan <kgryte@gmail.com>
